### PR TITLE
Plane Guided Alt Command: Use frame instead of bools for setting alt frame

### PR DESCRIPTION
**Pull request by Ryanf55**
# Purpose

When a plane receives an altitude command from mavlink [SET_POSITION_TARGET_GLOBAL_INT](https://mavlink.io/en/messages/common.html#SET_POSITION_TARGET_GLOBAL_INT), we were using the booleans in the Location field to set the frame. This switches to using the altitude setter with the frame which is more explicit. 

We now convert the frames with `mavlink_coordinate_frame_to_location_alt_frame`.

Conveniently, deleting code saves 40b flash.

# Tests performed

`./Tools/autotest/autotest.py test.Plane`. 

It seems like this might be covered by `SetpointGlobalPos`, but that test is not run by Plane.

I also used the `changealt`  and `changealt_abs` command in `GUIDED` mode while flying in SITL to verify we can change altitude. Those commands do not use this interface. 

I do not have a great way to test this as is. Suggestions welcome!

# Size Compare Durandal

```
Binary Name      Text [B]        Data [B]     BSS (B)      Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  --------------  -----------  -----------  ----------------------------  -------------------------
arducopter-heli  0 (0.0000%)     0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      14[28](https://github.com/ArduPilot/ardupilot/actions/runs/11563094404/job/32185663719?pr=28479#step:10:29)72
antennatracker   0 (0.0000%)     0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      615404
arducopter       0 (0.0000%)     0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      142344
ardurover        0 (0.0000%)     0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      [29](https://github.com/ArduPilot/ardupilot/actions/runs/11563094404/job/32185663719?pr=28479#step:10:30)6528
arduplane        -40 (-0.0022%)  0 (0.0000%)  0 (0.0000%)  -40 (-0.0022%)                                   145824
blimp            0 (0.0000%)     0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      593368
ardusub          0 (0.0000%)     0 (0.0000%)  0 (0.0000%)  0 (0.0000%)                                      352236
```

### CODE REVIEW

**Comment by Ryanf55** [comment]
@IamPete1  Mind taking a look? This is starting to move towards treating altitude frame booleans as private.

It will make this PR smaller: https://github.com/ArduPilot/ardupilot/pull/26328#discussion_r1522656642

File: ArduPlane/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -1382,22 +1382,20 @@ void GCS_MAVLINK_Plane::handle_set_position_target_global_int(const mavlink_mess
...
         {
-            cmd.content.location.alt = pos_target.alt * 100;
-            cmd.content.location.relative_alt = false;
-            cmd.content.location.terrain_alt = false;
+            const int32_t alt_cm = pos_target.alt * 100;
             switch (pos_target.coordinate_frame) 
             {
                 case MAV_FRAME_GLOBAL:
                 case MAV_FRAME_GLOBAL_INT:
+                    cmd.content.location.set_alt_cm(alt_cm, Location::AltFrame::ABSOLUTE);
```
Comment (peterbarker) [inline ArduPlane/GCS_Mavlink.cpp:1390]: If we're playing in here we really should move to `mavlink_coordinate_frame_to_location_alt_frame` instead.

Comment (Ryanf55) [reply ArduPlane/GCS_Mavlink.cpp:1390]: Can do!

**Comment by Ryanf55** [comment]
Switched to `mavlink_coordinate_frame_to_location_alt_frame` , now there's much less code. `changealt` still works.

File: ArduPlane/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -1382,33 +1382,19 @@ void GCS_MAVLINK_Plane::handle_set_position_target_global_int(const mavlink_mess
...
-                    break;
-            }    
+            const int32_t alt_cm = pos_target.alt * 100;
+            Location::AltFrame frame;
+            if (mavlink_coordinate_frame_to_location_alt_frame((MAV_FRAME)pos_target.coordinate_frame, frame)) {
+                gcs().send_text(MAV_SEVERITY_WARNING, "Invalid coord frame in SET_POSTION_TARGET_GLOBAL_INT");
+                msg_valid = false;
+            } else {
+                cmd.content.location.set_alt_cm(alt_cm, frame);
+            }
```
Comment (peterbarker) [inline ArduPlane/GCS_Mavlink.cpp:1392]:
```suggestion
                return;
            }
            cmd.content.location.set_alt_cm(alt_cm, frame);
```

also remove variable `msg_valid` entirely

**Review by peterbarker — CHANGES_REQUESTED** [review]
Please outline what testing you have done on this PR.

**Comment by Ryanf55** [comment]
> Please outline what testing you have done on this PR.

I did. Do you want me to upload logs, copy the exact terminal output, a video, or something more specific?

**Comment by Ryanf55** [comment]
I'm working on adding an autotest in `master` to preserve existing behavior. Once that's in, we can revisit this PR.
See https://github.com/ArduPilot/ardupilot/pull/28487

**Comment by peterbarker** [comment]
So PH and I weren't going to scope creep you on that bit :-)

OTOH, now that you've started - why not early-return on the alt-mask check too?  It's going to be a nice function after that.

**Comment by Ryanf55** [comment]
> So PH and I weren't going to scope creep you on that bit :-)
> 
> OTOH, now that you've started - why not early-return on the alt-mask check too? It's going to be a nice function after that.

If you have a way to easily test it and are comfortable. 

The reason I don't early return on the alt mask is there will be other masks to test against coming up. The alt can be controlled in the upcoming path guidance mode at the same time as other fields in this function.

**Review by IamPete1 — APPROVED** [review]
This looks correct to me.

**Pull request merged**